CONTRIBUTING.md

```diff
@@ -1,3 +1,540 @@
+
+You can do this by telling `pip` to use the official PyTorch Wheel Repository instead of the default PyPI repository.
+
+```
+$ pip install torch torchvision -f https://download.pytorch.org/whl/torch_stable.html
+```
+
+_Note_ If you need a specific version you can supply it with the `==` syntax. Also be aware there are different versions depending on if you require CUDA for GPU usage or CPU only such as what we use in GitHub CI.
+
+```
+$ pip install torch==1.7.1 torchvision==0.8.2 -f https://download.pytorch.org/whl/torch_stable.html
+```
+
+Then continue below with requirements.dev.txt like normal.
+
+**NOTE** this is required for several `dev` packages like pytest-xdist etc.
+
+```
+$ pip install -r requirements.dev.txt
+```
+
+or
+
+```
+$ pipenv install --dev --skip-lock
+```
+
+Now you can verify we have installed a lot of stuff by running:
+
+```
+$ pip freeze
+```
+
+### Linking the SyMPC src
+
+Now we need to link the src directory of the SyMPC code base into our site-packages
+so that it acts like it’s installed but we can change any file we like and `import` again
+to see the changes.
+
+```
+$ pip install -e .
+```
+
+The best way to know everything is working is to run the tests.
+
+Run the quick tests with all your CPU cores by running:
+
+```
+$ pytest -n auto
+```
+
+If they pass then you know everything is set up correctly.
+
+
+### Linking the PySyft src
+
+SyMPC is a companion library for PySyft. For the moment they are highly coupled.
+One of the SyMPC goals is to convert it into a standalone library. Thus, PySyft
+will depend on SyMPC, but not the other way around. This point is not yet reached
+and this adds some complications when applying some changes that in SyMPC that requires changes in PySyft to work.
+
+**Example**: You implement a new function on a ShareTensor in SyMPC. In your test, you call that function from a PySyft object pointer `share_pointer_in_pysyft.new_method()`.  Clearly, this test will not work until PySyft uses this new SyMPC version you are developing.
+
+To test this, you will need to clone the PySyft library, install the dependencies and link the PySyft src as we have done with SyMPC. For more detailed information, please visit the [PySyft contributin guide](https://github.com/OpenMined/PySyft/blob/dev/packages/syft/CONTRIBUTING.md).
+
+As a little summary, you need to:
+
+```bash
+$ git clone https://github.com/OpenMined/PySyft.git
+$ cd PySyft/packages
+$ pip install -r 'requirements.txt'
+$ pip install -e syft
+```
+
+This steps are here as a guide. Please, check the [PySyft contributin guide](https://github.com/OpenMined/PySyft/blob/dev/packages/syft/CONTRIBUTING.md) for the detailed steps.
+
+Now, you are prepared to face this kind of changes. A normal change in SyMPC looks like:
+
+1. Change SyMPC code
+2. Run the tests
+
+A change that implies both libraries looks like:
+
+1. Change SyMPC code
+2. Change PySyft code
+3. Install SyMPC with `pip install -e .`
```

Comment:
3 and 4 is optional - because if you already have the SyMPC and PySyft repos locally and you did the previous step, you don't need to do this every time you make a change.